src/lib/renderers/main-renderer.ts

```diff
@@ -0,0 +1,182 @@
+import type { IRenderer } from '@/components/webgpu-canvas';
+import type { SceneGraph } from '@/lib/scene';
+import type { WebGPUContext } from '@/lib/webgpu-context';
+
+export class MainRenderer implements IRenderer {
+  // ------------------------------------------------------------------------------------------
+  // ------ Setup: buffers, layouts, pipeline
+  // ------------------------------------------------------------------------------------------
+
+  // TODO: these might get killed since we don't have models per se
+  modelBindGroupLayout: GPUBindGroupLayout;
+  materialBindGroupLayout: GPUBindGroupLayout;
+
+  // TODO: these uniform guys
+  // sceneUniformsBindGroupLayout: GPUBindGroupLayout;
+  // sceneUniformsBindGroup: GPUBindGroup;
+
+  depthTexture: GPUTexture;
+  depthTextureView: GPUTextureView;
```

Comment:
@thumun @sgmq0 do we think these two guys should be initialized in `webgpu-context.ts` instead? they will likely be implemented in the `PreviewRenderer` as well

EDIT: hmm actually different renderers might not use depth depending on how they work, so it should prob be up to the specific renderer itself